### ISSUE TITLE
Log rotation not working on agents running on Redhat 64 bit el5 and el6

### ISSUE THREAD

**Issue by krobertson71** [NagiosEnterprises/ncpa#134]
I have checked our test windows servers and log rotation for the agent works.

However on Redhat el5 and el6 64bit it is not.

The config file is set to debug by default in ncpa.conf

-rwxrwxr-x 1 nagios nagcmd  85M Jan 15 11:56 ncpa_listener.log
-rw------- 1 root   root      4 Dec 10 22:39 ncpa_listener.pid
-rwxrwxr-x 1 nagios nagcmd 139M Jan 15 11:54 ncpa_passive.log
-rw------- 1 root   root      4 Dec 10 22:39 ncpa_passive.pid

As you can see there are getting quite large.

We have verified this on several Redhat servers.  This is going to be a problem for us.  HELP!

**Comment by scot0357** [comment]
Since this is such a severe issue I'm going to pull this in to the 1.8.0 release, which was frozen yesterday for testing.

**Comment by krobertson71** [comment]
I am not a dev so take this with a grain of salt.  I was looking at the code for the ncpadaemon.py and noticted you are calling hander operations (right term?) here in the below def.  I noticed you are calling stuff for 'from logging.handlers import RotatingFileHandler' but you are not importing the logging.handlers module.

RotatingFileHandler is from the module logging.handlers and I am noticing you are not importing that.

I looked at the Python documentation for 2.7 and it shows an example of how to handle log rotation and the code in that example imports logging.handlers.

def start_logging(self):
u"""Configure the logging module"""
try:
level = int(self.loglevel)
except ValueError:
level = getattr(logging, self.loglevel.upper())
handlers = []
if self.logfile:
if not self.logmaxmb:
handlers.append(logging.FileHandler(self.logfile))
else:
from logging.handlers import RotatingFileHandler
handlers.append(RotatingFileHandler(self.logfile, maxBytes=self.logmaxmb * 1024 * 1024, backupCount=self.logbackups))
self.chown(self.logfile)
handlers.append(logging.StreamHandler())
log = logging.getLogger()
log.setLevel(level)
for h in handlers:
h.setFormatter(logging.Formatter(
u"%(asctime)s %(process)d %(levelname)s %(message)s"))
log.addHandler(h)
